integration-tests/helpers/dataplane.go

```diff
@@ -70,6 +81,8 @@ func RunDataplane(t *testing.T, pod *Pod, suite *Suite, cfg DataplaneConfig) *Co
 			pod.MappedPorts[EnvoyAdminPort],
 		)
 
+		// TODO: This will fail because the integration test now performs a
+		// graceful shutdown of Envoy before reaching this point.
```

Comment:
The purpose of this, I think, is just to dump the envoy config for debugging, particularly in the case of a test failure?

One thought is to spawn a goroutine that periodically dumps the envoy config to a file? It would do more fetches and writes than necessary, but you'd at least have the latest version of the envoy config that it was able to fetch before the Envoy admin API became unavailable.

Better would be to somehow detect that Envoy is about to be (gracefully) shutdown, and dump the Envoy config at that point. But not sure how to easily accomplish that in this situation, and it may not be worth it.

Reply:
Added a proposed implementation of this in 476db3bf98497344179066b28bcc2df777b09bab